### PR TITLE
Login Page added

### PR DESCRIPTION
**Pull request by atharmohammad**
issue #21 

**Describe the changes you've made**

Added Login Page on Route /login and only when any account is logged in currently via google can access other routes and after user is logged in he can not access login page , currently just a testing loggedin variable is stored when logged in using google in localStorage , to test you can login using google and to logout just clear cache of your browser

you can see login page at route /login

**Describe if there is any unusual behavior (Any Warning) of your code(Write NA if there isn't)**

After logging in via google , I tried to redirect it to home using history.push('/home') after dispatch but redirect is not occuring
I tried it by async await but nothing happens , So currently if you want to go to '/home' you have to type mannualy after logging in, I am looking into this Tell me if there is a way

**Checklist**
- [X] My code follows the code style of this project.
- [X] The title of my pull request is a short description of the requested changes.

log

### CODE REVIEW

**Comment by atharmohammad** [comment]
@Abhishek-kumar09  can you look in changes and tell me what I have to change more

**Comment by KeenWarrior** [comment]
For now, we only wish to have Login with Google

**Comment by atharmohammad** [comment]
@KeenWarrior  I have made the changes , can you please review it 
log

File: src/Routes.js [hunk 1]
```diff
@@ -7,8 +7,9 @@ import HomePage from "./pages/HomeView";
 import CoursePage from "./pages/CoursePageViewWithVideo";
 import WishlistPage from "./pages/wishListView";
 import ProfilePage from "./pages/ProfilePage";
+import Login from './pages/Login/Login';
 
-const renderRoutes = () => (
+let renderRoutes = () => (
```
Comment (Abhishek-kumar09) [inline src/Routes.js:12]: Why this change?

File: src/pages/Login/Login.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  container:{
+    backgroundColor: theme.palette.secondary.main,
+    height: '100vh',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+    textAlign: 'center',
+    // minHeight: '100vh'
```
Comment (Abhishek-kumar09) [inline src/pages/Login/Login.js:33]: Please remove this.

File: src/pages/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+function Login(){
+  const user = useSelector(state => state.account.user);
+  const dispatch = useDispatch();
+  const classes = useStyles();
+  const handleLoginOpen =() => {
+    try{
+
+      dispatch(login());
+
+    }
```
Comment (Abhishek-kumar09) [inline src/pages/Login/Login.js:47]: Please fix unwanted new lines.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Some minor fixes for now.

File: src/Routes.js (hunk 1 above)
Comment (atharmohammad) [reply src/Routes.js:12]: I am Rendering all routes only if a person is logged in , so I have to change renderRoutes if a person is nnot logged in that is why i make it a let instead of const

**Comment by atharmohammad** [comment]
@Abhishek-kumar09  I have made changes tell me if any thing you want me to change

File: src/services/authService.js [hunk 4]
```diff
@@ -52,9 +52,12 @@ class AuthService {
   }
 
   login = () => {
-    cfaSignIn("google.com").subscribe((user) => {
-      console.log(user);
-    });
+    cfaSignIn('google.com').subscribe(
+       (user) => {
+        console.log(user);
+        localStorage.setItem('loggedin','loggedintoken'); //For test loggedin is stored
```
Comment (Abhishek-kumar09) [inline src/services/authService.js:58]: you are setting `loggedin` to true when user is logging in, but no changes in this value when the user logged out, So if user logged In and then he logged out, there will be `loggedIn` state still true, which is not ideal.

File: src/Routes.js [hunk 5]
```diff
@@ -61,6 +62,14 @@ const renderRoutes = () => (
...
 
 function Routes() {
+  if(localStorage.getItem('loggedin') == null){ //In place of logged in token will be stored
+    renderRoutes = ()=>(
+      <Switch>
+        <Route path='/login' component={Login}/>
+        <Redirect to='/login'/>
+      </Switch>
+    )
+  }
```
Comment (Abhishek-kumar09) [inline src/Routes.js:72]: This is not a good approach. We should not rely upon local storage.
Since we are using firebase and redux you can fetch the state of the user with `useSelector` hook, and if user ispresent navigate to home else login.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/Routes.js (hunk 5 above)
Comment (Abhishek-kumar09) [reply src/Routes.js:72]: Instead of creating render routes here, you can do something like: 
```
if user 
	history.push("/home")
else 
	history.push("/login")
```

**Comment by KeenWarrior** [comment]
@atharmohammad Hey I am closing this PR in regards to #68 

Login was becoming a roadblock for other stuff.

Kudos for your work though. Directly took your Login.js 🥳

**Comment by atharmohammad** [comment]
> @atharmohammad Hey I am closing this PR in regards to #68 
> 
> Login was becoming a roadblock for other stuff.
> 
> Kudos for your work though. Directly took your Login.js 🥳 

Okay,I just couldn't push the changes due to error I was getting due to eslint and prettier after the pull
